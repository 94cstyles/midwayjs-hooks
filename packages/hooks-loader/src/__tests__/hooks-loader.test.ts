import { resolve } from 'path'
import compiler from './compiler'
import webpack from 'webpack'
import { wrap } from 'jest-snapshot-serializer-raw'

const root = resolve(__dirname, './fixtures/catch-all')

const resolveEntry = (path: string | string[]) => {
  if (!Array.isArray(path)) {
    return resolve(__dirname, root, 'src/apis/', path)
  }
  return path.map((p) => resolve(__dirname, root, 'src/apis/', p))
}

const getOutput = (stats: webpack.Stats) => {
  return stats.toJson().modules[2].source
}

describe('hooks-loader', () => {
  test('Compile render', async () => {
    const stats = await compiler(resolveEntry('render/[...index].ts'), root)
    const output = getOutput(stats)
    expect(wrap(output)).toMatchSnapshot()
  })

  test('Compile lambda', async () => {
    const stats = await compiler(resolveEntry('lambda/index.ts'), root)
    const output = getOutput(stats)
    expect(wrap(output)).toMatchSnapshot()
  })

  test('the second build should match the first.', async () => {
    const first = await compiler(resolveEntry('lambda/index.ts'), root)
    const second = await compiler(resolveEntry('lambda/index.ts'), root)
    expect(getOutput(first)).toEqual(getOutput(second))
  })

  test('non-lambda files should not be compiled', async () => {
    const stats = await compiler(resolveEntry('util/util.ts'), root)
    expect(wrap(stats.toJson().modules[0].source)).toMatchSnapshot()
  })
})
