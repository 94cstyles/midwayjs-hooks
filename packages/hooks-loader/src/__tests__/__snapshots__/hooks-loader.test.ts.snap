// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hooks loader with proxy Compile lambda 1`] = `
import { createRequest } from '@midwayjs/hooks-core/lib/request/sdk';

export default createRequest('/api', 'default');
`;

exports[`hooks loader with proxy Compile render 1`] = `
import { createRequest } from '@midwayjs/hooks-core/lib/request/sdk';

export default createRequest('/', 'default');
export const foo = createRequest('/', 'foo');
export const bar = createRequest('/', 'bar');
`;

exports[`hooks loader with proxy non-lambda files should not be compiled 1`] = `
export function isTrue(value) {
    return value === 'true' || value === true;
}

`;
